torch/csrc/jit/pickler.cpp

```diff
@@ -115,6 +115,17 @@ void Pickler::pushIValueImpl(const IValue& ivalue) {
     push<PickleOpCode>(PickleOpCode::BUILD);
   } else if (ivalue.isDevice()) {
     pushDevice(ivalue);
+  } else if (ivalue.isCapsule()) {
+    std::stringstream err;
+    err << "Cannot serialize custom bound C++ class";
+    if (memorized_class_types_ && memorized_class_types_->size()) {
+      if (auto qualname = memorized_class_types_->back()->name()) {
```

Comment:
are you sure that the last one is guaranteed to be current class? If so - I'd add a comment about it

Reply:
It's guaranteed to the be current class so long as we only ever see a `Capsule` as an attribute of an enclosing `Object`

Reply:
```
ObjectA {
   ObjectB b;
   Capusle c;
}
```
Capsule will get attributed to ObjectB

Reply:
That's only possible if a user can explicitly put a Capsule in a class, which I don't think is possible